common/helpers/g2p.py

```diff
@@ -0,0 +1,101 @@
+__all__ = ["G2PBasedOovAugmenter"]
+
+import os
+from typing import Optional
+
+from sisyphus import tk
+Path = tk.setup_path(__package__)
+
+from i6_core.corpus.stats import ExtractOovWordsFromCorpusJob
+from i6_core.g2p.apply import ApplyG2PModelJob
+from i6_core.g2p.convert import BlissLexiconToG2PLexiconJob, G2POutputToBlissLexiconJob
+from i6_core.g2p.train import TrainG2PModelJob
+
+
+class G2PBasedOovAugmenter():
+    def __init__(self,
+                 original_bliss_lexicon:str,
+                 train_lexicon:Optional[str] = None,
+                 g2p_model_path:Optional[Path] = None,
+                 train_args:Optional[dict] = None,
+                 apply_args:Optional[dict] = None,
+                 ):
+        """
+        :param original_bliss_lexicon: path to the original lexicon that will be augmented with OOVs
+        :param train_lexicon: path to the train lexicon in case it differs from the original lexicon
+        :param g2p_model_path: path to the g2p model, if None a g2p model is trained
+        #######################################
+        :param train_args = {
+        "num_ramp_ups"   :4,
+        "min_iter"       :1,
+        "max_iter"       :60,
+        "devel"          :"5%",
+        "size_constrains":"0,1,0,1"
+        }
+        #######################################
+        :param apply_args = {
+        "variants_mass"  :1.0,
+        "variants_number":1
+        }
+        """
+        super().__init__()
+        self.original_bliss_lexicon = original_bliss_lexicon
+        self.train_lexicon = original_bliss_lexicon if train_lexicon is None else train_lexicon
+        self.g2p_model_path = g2p_model_path
+        self.train_args = {}
+        self.apply_args = {}
+
+        if train_args is not None:
+            self.train_args.update(train_args)
+
+        if apply_args is not None:
+            self.apply_args.update(apply_args)
+
+
+    def _train_and_set_g2p_model(self, alias_path:str):
+        g2p_lexicon_job = BlissLexiconToG2PLexiconJob(bliss_lexicon=self.train_lexicon)
+
+        g2p_train_job = TrainG2PModelJob(
+            g2p_lexicon=g2p_lexicon_job.out_g2p_lexicon,
+            **self.train_args)
+        g2p_train_job.add_alias(os.path.join(alias_path, "train_g2p_model"))
+        self.g2p_model_path = g2p_train_job.out_best_model
+
+
+    def get_g2p_augmented_bliss_lexicon(
+            self,
+            bliss_corpus:Path,
+            corpus_name:str,
+            alias_path:str
+    ):
+        extract_oov_job = ExtractOovWordsFromCorpusJob(
+            bliss_corpus=bliss_corpus,
+            bliss_lexicon=self.original_bliss_lexicon
+        )
+        extract_oov_job.add_alias(os.path.join(alias_path, "extract-oov-from-{}".format(corpus_name)))
+
+        if self.g2p_model_path is None:
```

Comment:
This can be part of the init, so that you do not call this multiple times.

Reply:
This is done here, because it needs the `alias_path`. I do want the class having this path. If you prefer I can avoid adding the alias. Otherwise, this is where it should be called.

Reply:
Hmm I see, I am not fully happy with this but then it is okay.